services/supervisor/src/transactions.rs

```diff
@@ -92,26 +92,43 @@ impl StartService {
         let dispatcher_data = context.data().for_dispatcher();
 
         // Check that artifact is deployed.
-        if dispatcher_data
+        dispatcher_data
             .get_artifact(self.artifact.name.as_str())
-            .is_none()
-        {
-            log::trace!(
-                "Discarded start of service {} from the unknown artifact {}.",
-                &self.name,
-                &self.artifact.name,
-            );
-
-            return Err(Error::UnknownArtifact.into());
-        }
+            .ok_or_else(|| {
+                Error::UnknownArtifact.with_description(format!(
+                    "Discarded start of service {} from the unknown artifact {}.",
+                    &self.name, &self.artifact.name,
+                ))
+            })?;
 
         // Check that there is no instance with the same name.
         if dispatcher_data.get_instance(self.name.as_str()).is_some() {
-            log::trace!(
+            return Err(Error::InstanceExists.with_description(format!(
                 "Discarded start of the already running instance {}.",
                 &self.name
-            );
-            return Err(Error::InstanceExists.into());
+            )));
+        }
+
+        Ok(())
+    }
+}
+
+impl StopService {
+    fn validate(&self, context: &CallContext<'_>) -> Result<(), ExecutionError> {
+        let instance = context
+            .data()
+            .for_dispatcher()
+            .get_instance(self.instance_id)
+            .ok_or_else(|| {
+                Error::MalformedConfigPropose
```

Comment:
Nit: I wonder if it could make sense to split `MalformedConfigPropose` to more fine-grained variants. Could be done separately.

Reply:
ECR-3984